### ISSUE TITLE
issues with fiel names

### ISSUE THREAD

**Issue by montenegrina** [autism-research-centre/SSC_liftover_imputation#1]
Hello,

I am doing this part of your code:

i in {1..22}; do ./plink --bfile ./imputed_plinkfile/1Mv1imputed_chr${i} --extract ./Imputed_1000G/chr${i}exclude.txt --make-bed --maf 0.05 --geno 0.05 --hwe 0.000001 --out ./imputed_plinkfile/1Mv1_chr${i}; done
      
./plink --bfile 1Mv1_chr22 --merge-list 1Mv1mergelist.txt --make-bed -biallelic-only --out 1Mv1_merged

for i in {1..22}; do ./plink --bfile 1Mv1_chr${i} --exclude 1Mv1_merged-merge.missnp --make-bed --out 1Mv1_chr${i}_2; done

./plink --bfile 1Mv1_chr22_2 --merge-list 1Mv1mergelist2.txt --make-bed -biallelic-only --out 1Mv1_merged

./plink --bfile 1Mv1_merged --maf 0.05 --update-name ~/SFARI/liftOverPlink/plinkrecodingfile.txt --hwe 0.000001 --geno 0.05 --mind 0.05 --make-bed --out 1Mv1_mergedQC

And it is unclear to me where do these files come from: 1Mv1imputed_chr${i}
is it rather  1Mv1_chr${i}?

Also why do in that line: --extract chr${i}exclude.txt when it was done above in :
for i in {1..22}; do ./plink --vcf ./Imputed/chr${i}.dose.vcf.gz --make-bed --out ./imputed_plinkfile/1Mv1_chr${i}  --extract ./Imputed/chr${i}exclude.txt --const-fid 0; done

Please advise,
Ana

**Comment by vwarrier** [comment]
Hi Ana,

"And it is unclear to me where do these files come from: 1Mv1imputed_chr${i}
is it rather 1Mv1_chr${i}?"

You're right. It's Mv1_chr${i}. That was a typo on my part whilst copying the code onto github.

You're also right that you don't need to extract it again. You can very well combine both the lines of code into one by adding --make-bed --maf 0.05 --geno 0.05 --hwe 0.000001 into the previous one.

This is very much only a framework of the code, so feel free to adapt it as you please. 

Best
Varun

**Comment by montenegrina** [comment]
Hi Varun,

Thank you so much for getting back to me, so if I understand you correctly
this would be the line:

for i in {1..22}; do ./plink --vcf ./Imputed/chr${i}.dose.vcf.gz
--make-bed --out ./imputed_plinkfile/1Mv1_chr${i}  --extract
./Imputed/chr${i}exclude.txt --geno 0.05 --hwe 0.000001 --const-fid 0;
done

and this one is not needed:
i in {1..22}; do ./plink --bfile
./imputed_plinkfile/1Mv1imputed_chr${i} --extract
./Imputed_1000G/chr${i}exclude.txt --make-bed --maf 0.05 --geno 0.05
--hwe 0.000001 --out ./imputed_plinkfile/1Mv1_chr${i}; done

Also I am not seeing in your code how do you deal with duplicated SNPs
after merning...I am having issues with that when I try to do
--update-name

I tried a couple of methods prior --update-name

plink --bfile 1Mv1_merged --write-snplist --out all_snps
cat all_snps.snplist | sort | uniq -d > duplicated_snps.snplist
plink --bfile 1Mv1_merged --exclude duplicated_snps.snplist --make-bed
1Mv1_merged_noDups

plink2 --bfile 1Mv1_merged --rm-dup retain-mismatch --make-bed --out
MyDataLessDuplicates
plink2 --bfile 1Mv1_merged -rm-dup list
plink --bfile 1Mv1_merged --exclude plink2.rmdup.mismatch --make-bed
--out 1Mv1_merged_noDup

but none of them works. Do you know about any comprehensive POST
Impute pipeline? I did imputation on Michigan Imputation Server.

Also is
Rsq > 0.6 and ALT_Frq > 0.05 & ALT_Frq < 0.95 standard in your opinion
and do you do any pruning afterwards?

Cheers,

Ana

**Comment by vwarrier** [comment]
Hi Ana,


I probably would have just removed it using R I think (remove duplicates). I can't quite remember, but I think plink may remove duplicates for you automatically...


Re. post-imputation, I have created another, more-up-to date pipeline that uses the TOPMED imputation server. I would suggest using that for your analyses (though please amend as needed). The link is here: https://github.com/vwarrier/ABCD_geneticQC


Re. the maf and the imputation r2, the values I've used here is a bit conservative. I think anything with an imputation r2 > 0.4 is good. With MAF, it's your call. Perhaps you can go as low at 1%, but not sure how well that would work with a sample size that's small (similar to that in SSC). I don't do any further pruning for basic gwas, though for polygenic scores you may want to clump depending on the methods you're using.


Best

Varun

?

________________________________

**Comment by montenegrina** [comment]
Thank you so much! Just for the reference can you please tell me how many
subjects/variants your ABCD cohort has?
I am really having trouble deciding on MAF. I have 1640 subjects and
imputed around 6 million SNPs (before imputation I had around 360 000)
In my case what you would this same range for MAF be optimal?
ALT_Frq > 0.001 & ALT_Frq < 0.999

**Comment by vwarrier** [comment]
Hi Ana,


You can always run a more stringent maf filter later downstream. I'd suggest sticking to a minimum maf of 1% for a cohort of 1640 subjects. This means that at least 16 participants will have the minor allele.


Best

Varun

​

________________________________

**Comment by montenegrina** [comment]
Hi Varun,

thank you so much for that tip! I found this analysis here:
https://www.mv.helsinki.fi/home/mjxpirin/GWAS_course/material/GWAS3.html
and for my sample size it suggests that my MAF should be 0.5 in order to
have 65% power...

And one more question do you think I can use PCs I calculated prior
imputation (for non imputed, but QCd data) in order to do GWAS on imputed
data?

Cheers,
Ana

**Comment by vwarrier** [comment]
Yes, PCs using QCd genotyped (but not imputed data) should be fine.

Best

Varun

​

________________________________

**Comment by montenegrina** [comment]
Thank you so much for all your help!
